perfkitbenchmarker/benchmarks/mysql_service_benchmark.py

```diff
@@ -0,0 +1,773 @@
+    # We need to keep querying the status of the deletion here before we let
+    # this go. RDS DB deletion takes some time to finish. And we have to
+    # wait until this DB is deleted before we proceed because this DB holds
+    # references to various other resources: subnet groups, subnets, vpc, etc.
+    delete_db_cmd = AWS_PREFIX + [
+        'rds',
+        'delete-db-instance',
+        '--db-instance-identifier', vm.db_instance_id,
+        '--skip-final-snapshot']
+
+    logging.info('Will call this command to delete db instance: \n %s',
+                 ' '.join(delete_db_cmd))
+
+    stdout, stderr, _ = vm_util.IssueCommand(delete_db_cmd)
+    logging.info('Request to delete the DB has been issued, stdout:\n%s\n'
+                 'stderr:%s\n', stdout, stderr)
+
+    status_query_cmd = AWS_PREFIX + [
+        'rds',
+        'describe-db-instances',
+        '--db-instance-id', vm.db_instance_id]
+    status_query_count = 1
+    while True:
+      try:
+        response = json.loads(stdout)
+        db_status = _RDSParseDBInstanceStatus(response)
+        if db_status == 'deleting':
+          status_query_count += 1
+          if status_query_count > DB_STATUS_QUERY_LIMIT:
+            logging.warn('DB is still in deleting state after long wait, bail.')
+            break
+
+          # Wait for a few seconds and query status
+          time.sleep(DB_STATUS_QUERY_INTERVAL)
+          stdout, stderr, _ = vm_util.IssueCommand(status_query_cmd)
+        else:
+          break
+      except:
+        # stdout cannot be parsed into json, it might simply be empty -
+        # deletion has completed.
+        break
+
+    # Outside the loop, look at stderr, expect it to have the
+    # "DB instance Not found" string.
+    # if it does not have, log a warning but proceed forward.
+    if re.findall('DBInstanceNotFound', stderr):
+      logging.info('DB has been successfully deleted, got confirmation.')
+    else:
+      logging.warn('DB may not have been deleted, we did not get confirmation '
+                   'from stderr, which is %s', stderr)
+
+    if hasattr(vm, 'db_subnet_group_name'):
+      delete_db_subnet_group_cmd = AWS_PREFIX + [
+          'rds',
+          'delete-db-subnet-group',
+          '--db-subnet-group-name', vm.db_subnet_group_name]
+      stdout, stderr, _ = vm_util.IssueCommand(delete_db_subnet_group_cmd)
+      logging.info('Deleted the db subnet group. stdout is:\n%s, stderr: \n%s',
+                   stdout, stderr)
+
+    if hasattr(vm, 'extra_subnet_for_db'):
+      delete_extra_subnet_cmd = AWS_PREFIX + [
+          'ec2',
+          'delete-subnet',
+          '--subnet-id=%s' % vm.extra_subnet_for_db.id]
+      stdout, stderr, _ = vm_util.IssueCommand(delete_extra_subnet_cmd)
+      logging.info('Deleted the extra subnet. stdout is:\n%s, stderr: \n%s',
+                   stdout, stderr)
+
+
+class GoogleCloudSQLBenchmark(object):
```

Comment:
[Similar to my explanation above] These preparation steps and logic are specific to this benchmark which deals with managed mysql service, no other benchmarks need these code, so it's better we keep them here, it makes the code much easier to refer to and read. (This is also consistent with how we organize other benchmarks, e.g., downloading gsutil for object storage benchmark which is specific to object storage benchmark so that logic is contained in the object storage benchmark file, rather than being placed into the common/shared GCP directory)